File: software/examples/csharp/ExampleSimple.cs
using System;
using Tinkerforge;

class Example
{
	private static string HOST = "localhost";
	private static int PORT = 4223;
	private static string UID = "XYZ"; // Change XYZ to the UID of your Accelerometer Bricklet 2.0

	static void Main()
	{
		IPConnection ipcon = new IPConnection(); // Create IP connection
		BrickletAccelerometerV2 a =
		  new BrickletAccelerometerV2(UID, ipcon); // Create device object

		ipcon.Connect(HOST, PORT); // Connect to brickd
		// Don't use device before ipcon is connected

		// Get current acceleration
		int x, y, z;
		a.GetAcceleration(out x, out y, out z);

		Console.WriteLine("Acceleration [X]: " + x/10000.0 + " g");
		Console.WriteLine("Acceleration [Y]: " + y/10000.0 + " g");
		Console.WriteLine("Acceleration [Z]: " + z/10000.0 + " g");

		Console.WriteLine("Press enter to exit");
		Console.ReadLine();
		ipcon.Disconnect();
	}
}

File: software/examples/rust/example_simple.rs
use std::{error::Error, io};

use tinkerforge::{accelerometer_v2_bricklet::*, ip_connection::IpConnection};

const HOST: &str = "localhost";
const PORT: u16 = 4223;
const UID: &str = "XYZ"; // Change XYZ to the UID of your Accelerometer Bricklet 2.0.

fn main() -> Result<(), Box<dyn Error>> {
    let ipcon = IpConnection::new(); // Create IP connection.
    let a = AccelerometerV2Bricklet::new(UID, &ipcon); // Create device object.

    ipcon.connect((HOST, PORT)).recv()??; // Connect to brickd.
                                          // Don't use device before ipcon is connected.

    // Get current acceleration.
    let acceleration = a.get_acceleration().recv()?;

    println!("Acceleration [X]: {} g", acceleration.x as f32 / 10000.0);
    println!("Acceleration [Y]: {} g", acceleration.y as f32 / 10000.0);
    println!("Acceleration [Z]: {} g", acceleration.z as f32 / 10000.0);

    println!("Press enter to exit.");
    let mut _input = String::new();
    io::stdin().read_line(&mut _input)?;
    ipcon.disconnect();
    Ok(())
}
